src/lib/components/Widget.tsx

```diff
@@ -77,19 +77,7 @@ function Updaters() {
   )
 }
 
-export type WidgetProps<T extends JSXElementConstructor<any> | undefined = undefined> = {
-  theme?: Theme
-  locale?: SupportedLocale
-  provider?: EthProvider
-  jsonRpcEndpoint?: string
-  width?: string | number
-  dialog?: HTMLElement | null
-  className?: string
-  onError?: ErrorHandler
-} & (T extends JSXElementConstructor<any>
-  ? ComponentProps<T>
-  : // eslint-disable-next-line @typescript-eslint/ban-types
-    {})
+export type WidgetProps = SwapProps
```

Comment:
These props should stay in `Widget` to make it easier when we build add'l widget components.

Reply:
I was running into some weird/major ts drama trying to make it work w/ the prior setup. My plan here is to eventually make this something like `type WidgetProps = SwapProps | PriceProps | OtherWidgetProps`